lib/modules/manager/bazel/extract.spec.ts

```diff
@@ -87,6 +87,29 @@ describe('modules/manager/bazel/extract', () => {
       ]);
     });
 
+    it('extracts dependencies for oci_pull deptype', () => {
+      const res = extractPackageFile(
+        codeBlock`
+          oci_pull(
+            name="hasura",
+            image="index.docker.io/hasura/graphql-engine",
+            # v1.0.0-alpha31.cli-migrations 11/28
+            digest="sha256:a4e8d8c444ca04fe706649e82263c9f4c2a4229bc30d2a64561b5e1d20cc8548",
+            tag="v1.0.0-alpha31.cli-migrations"
```

Comment:
fyi, `oci_pull` doesn't allow setting both `tag` and `digest`

Reply:
I can split the test into two (one using only digest, one using only tag). Do you think the parsing implemented in this should reject images that have both?

Reply:
Not really. oci_pull already fails when you set both. so it's unlikely that anyone will push broken code, therefore parser should not care at all.

Reply:
I am curious how would the parser find a newer version of digests? based on a tag? if yes, where would it get the tag information since we don't allow setting both?

Reply:
I believe if no tag is set, the datasource will implicitly query for the `latest` tag.
Maybe `rules_oci` should allow for a tag to allow pinning it for tools like renovate and simply ignore it if a hash is present?

Reply:
is that the behavior rules_docker has in renovate? I could see us allowing both tag and digest to be set and making digest override tag.

Reply:
This is the behavior I observed so far. I cannot really comment on the details of the docker datasource in renovate. Maybe someone else can confirm this.
What I do know is that a tag, if set, can be used for more informed upgrades.
Users can even define custom policies and versioning schemes to have fine control over what upgrades should be suggested by renovate.

Reply:
I see. because every image might not have the latest tag. also, keep in mind that the public API for oci_pull is not finalized yet. we have plans to change it.

Reply:
As long as the basic structure stays the same * I offer to maintain the renovate support and keep it compatible.

`*` by that I mean pull rules have some form of image name and a combination of tag and digest.

Reply:
I'll create an issue on rules_oci to track the potential api change of allowing a tag and a digest in one rule.

Reply:
@thesayyn should we merge this or would you prefer to wait until the api is completely stabilized?

Reply:
if you need the support right away i don't see an issue. i just wanted to give a heads-up